src/utils/delaySending.js

```diff
@@ -0,0 +1,28 @@
+const { promisify } = require('util');
+
+const sleep = promisify(setTimeout);
+
+/**
+ *
+ * This function will send out one request per second, until all requests have
+ * been sent.
+ *
+ * @param {Function} sender function that needs to be delayed
+ * @param {Array} items an array of items to iterate over and process through the sender
+ * @param {Number} ms a time in miliseconds that each request should wait for before processing
+ */
+const delaySending = async function delaySending(sender, items, ms) {
+  // eslint-disable-next-line no-const-assign
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < items.length; i++) {
```

Comment:
Nice fix to serialize the async tasks! Do you like `for...of` loops better? Either way 👍

Reply:
I find this more readable, but that's because I rarely ever right actual for loops. Always finding a way to use those array prototypes 😁. Would you prefer a for of here?

Reply:
@therynamo All good.